.github/workflows/publish-charts.yml

```diff
@@ -6,13 +6,71 @@ on:
       version:
         description: "Temporal version to use for release (for example 1.22.4)"
         required: true
+      bump:
+        description: "Chart version bump level"
+        required: true
+        type: choice
+        default: minor
+        options:
+          - minor
+          - patch
 
 jobs:
   release:
     permissions:
       contents: write
     runs-on: ubuntu-latest
     steps:
-      - name: Empty
+      - name: Generate token
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.TEMPORAL_CICD_APP_ID }}
+          private-key: ${{ secrets.TEMPORAL_CICD_PRIVATE_KEY }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
+          token: ${{ steps.generate_token.outputs.token }}
+          ref: rh-chart-publish
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Configure Git
         run: |
-          echo Doing nothing
+          git config --local user.name 'Temporal Data'
+          git config --local user.email 'commander-data@temporal.io'
+      
+      - name: Update Temporal chart version
+        working-directory: charts/temporal
+        run: |
+          RE='\([0-9]*\)\.\([0-9]*\)\.\([0-9]*\)'
+
+          current="$(grep -m 1 '^version:' Chart.yaml | awk '{print $2}')"
+
+          MAJOR=`echo $current | sed -e "s#$RE#\1#"`
+          MINOR=`echo $current | sed -e "s#$RE#\2#"`
+          PATCH=`echo $current | sed -e "s#$RE#\3#"`
+
+          if [ "${{ github.event.inputs.bump }}" == "minor" ]; then
+            let MINOR+=1
+            let PATCH=0
+          else
+            let PATCH+=1
+          fi
+          NEW_VERSION="$MAJOR.$MINOR.$PATCH"
+
+          sed -i -e "s/^version: .*/version: $NEW_VERSION/" Chart.yaml
+          sed -i -e "s/^appVersion: .*/appVersion: ${{ github.event.inputs.version }}/" Chart.yaml
+
+          git commit -m "Update Chart to $NEW_VERSION, Temporal v${{ github.event.inputs.version }}" Chart.yaml
+          git push origin
```

Comment:
This is really nice; I learned a couple of things from it; `grep -m` and I didn't know about the `let` arithmetic at all, I thought one had to use `((...))`. Could use `sed -E` to make the regex more readable?)